libdevcore/Worker.cpp

```diff
@@ -50,11 +50,16 @@ void Worker::startWorking()
 //				cnote << "Trying to set Started: Thread was" << (unsigned)ex << "; " << ok;
 				(void)ok;
 
-				startedWorking();
-//				cnote << "Entering work loop...";
-				workLoop();
-//				cnote << "Finishing up worker thread...";
-				doneWorking();
+				try
+				{
+					startedWorking();
+					workLoop();
+					doneWorking();
+				}
+				catch (std::exception const& _e)
```

Comment:
what about boost exceptions? Note that boost::current_exception_diagnostic_information() works for both